src/api/AuthenticationManagement.ts

```diff
@@ -33,29 +39,173 @@ export default class AuthenticationManagement extends Common {
                 result.statusCode = response.status;
                 console.log(response);
             })
-            .catch((error: AxiosError) => {
+            .catch(async (error: AxiosError) => {
                 if (error.response) {
                     result.statusCode = error.response.status;
                     result.error = error.response.data as ValidationError;
+                    reloginSuccess = await handleAuthenticationError(result);
                 } else {
                     result.networkError = true;
                 }
             });
 
+        if (result.statusCode == 401 && reloginSuccess) {
+            return await this.changeOwnPassword(password);
+        }
+
         if (result.returnValue) {
             const store = useStore();
-            store.commit(MutationTypes.SET_LOGINDATA, { username: username, password: password });
             store.commit(MutationTypes.SET_LOGGEDIN, true);
         }
 
         return result;
     }
+
+    /**
+     * Authenticate against Lagom endpoint, return true if successful
+     * @param loginData
+     */
+    static async login(loginData?: { username: string; password: string }): Promise<APIResponse<boolean>> {
+        if (loginData !== undefined) {
+            return await this._getRefreshToken(loginData);
+        } else {
+            return await this._getLoginToken();
+        }
+    }
+
+    static async _getLoginToken(): Promise<APIResponse<boolean>> {
+        const instance = axios.create({
+            baseURL: process.env.VUE_APP_API_BASE_URL + "/authentication-management",
+            headers: {
+                "Accept": "*/*",
+                "Content-Type": "application/json;charset=UTF-8",
+            },
+            withCredentials: true,
+        });
+
+        let result: APIResponse<boolean> = {
+            error: {} as APIError,
+            networkError: false,
+            returnValue: false,
+            statusCode: 0,
+        };
+
+        let username: string = "";
+
+        await instance
+            .get(`/refresh`)
+            .then((response: AxiosResponse) => {
+                result.statusCode = response.status;
+                result.returnValue = true;
+                username = response.data.username;
+            })
+            .catch((error: AxiosError) => {
+                if (error.response) {
+                    result.statusCode = error.response.status;
+                } else {
+                    result.networkError = true;
+                }
+            });
+
+        if (result.returnValue) {
+            const store = useStore();
+
+            store.commit(MutationTypes.SET_LOGGEDIN, true);
+
+            const userManagement = new UserManagement();
+            const handler = new GenericResponseHandler();
+            const response = await userManagement.getSpecificUser(username);
+            const user = handler.handleResponse(response);
+            store.commit(MutationTypes.SET_USER, user);
```

Comment:
What if the API call fails? Then we should not commit the user, as we do not have one, should we?